Makefile

```diff
@@ -568,7 +572,7 @@ $(STATIC_NAME): $(OBJS) | $(LIB_BUILD_DIR)
 	@ echo AR -o $@
 	$(Q)ar rcs $@ $(OBJS)
 
-$(BUILD_DIR)/%.o: %.cpp | $(ALL_BUILD_DIRS)
+$(BUILD_DIR)/%.o: %.cpp | $(ALL_BUILD_DIRS) $(CUB_DIR)
```

Comment:
Is CUB going to be needed for (non-CUDA) object files? It would nice if this line were not necessary, so that CUB does not need to be downloaded for (e.g.) CPU-only builds.